yt_dlp/extractor/megatvcom.py

```diff
@@ -0,0 +1,190 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+from urllib.parse import (
+    parse_qs,
+    urlparse,
+)
+
+from .common import InfoExtractor
+from ..utils import (
+    clean_html,
+    determine_ext,
+    ExtractorError,
+    extract_attributes,
+    get_element_by_class,
+    HEADRequest,
+    unescapeHTML,
+    unified_timestamp,
+)
+
+
+class MegaTVComBaseIE(InfoExtractor):
+    _PLAYER_DIV_ID = 'player_div_id'
+
+    def _extract_player_attrs(self, webpage):
+        PLAYER_DIV_RE = r'''(?x)
+        <div(?:
+            id=(?P<_q1>["'])(?P<%(pdi)s>%(pdi)s)(?P=_q1)|
+            [^>]*?
+        )+>
+        ''' % {'pdi': self._PLAYER_DIV_ID}
+        for mobj in re.finditer(PLAYER_DIV_RE, webpage):
+            if mobj.group(self._PLAYER_DIV_ID):
+                player_el = mobj.group(0)
+                break
+        else:
+            raise ExtractorError('no <div id="%s"> element found in webpage' %
+                                 self._PLAYER_DIV_ID)
+        return {
+            re.sub(r'^data-(?:kwik_)?', '', k): v
+            for k, v in extract_attributes(player_el).items()
+            if k not in ('id',)
+        }
+
+
+class MegaTVComIE(MegaTVComBaseIE):
+    IE_NAME = 'megatvcom'
+    IE_DESC = 'megatv.com videos'
+    _VALID_URL = r'https?://(?:www\.)?megatv\.com/(?:(?!\d{4})[^/]+/(?P<id>\d+)|\d{4}/\d{2}/\d{2})/(?P<_slug>[^/]+)'
+
+    _TESTS = [{
+        'url': 'https://www.megatv.com/2021/10/23/egkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia/',
+        'md5': '6546a1a37fff0dd51c9dce5f490b7d7d',
+        'info_dict': {
+            'id': '520979',
+            'ext': 'mp4',
+            'title': 'md5:70eef71a9cd2c1ecff7ee428354dded2',
+            'description': 'md5:0209fa8d318128569c0d256a5c404db1',
+            'timestamp': 1634975747,
+            'upload_date': '20211023',
+        },
+    }, {
+        'url': 'https://www.megatv.com/tvshows/527800/epeisodio-65-12/',
+        'md5': 'cba2085d45c1abeb8e7e9b7e1d6c0072',
+        'info_dict': {
+            'id': '527800',
+            'ext': 'mp4',
+            'title': 'md5:fc322cb51f682eecfe2f54cd5ab3a157',
+            'description': 'md5:b2b7ed3690a78f2a0156eb790fdc00df',
+            'timestamp': 1636048859,
+            'upload_date': '20211104',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        display_id = self._match_valid_url(url).group('_slug')
+        _is_article = video_id is None
+        webpage = self._download_webpage(url,
+                                         display_id if _is_article
+                                         else video_id)
+        if _is_article:
+            video_id = self._search_regex(
+                r'<article(?:id=(?P<_q>["\'])Article_(?P<article>\d+)(?P=_q)|[^>]*?)+>',
+                webpage, 'article_id', group='article'
+            )
```

Comment:
```suggestion
            video_id = self._search_regex(
                r'<article[^>]*\sid=["\']Article_(\d+)["\']', webpage, 'article id')
```
the regex is unnecessarily complicated

Reply:
The regex can be further simplified, but I am trying to use a consistent style for parsing attributes whatever order they may appear in, and when looking for more than one attribute `[^>]*\s` doesn't work, IIRC.
Skipping the group requires not using the named group (`_q`) for the matched quote, and I think matching the same closing quote char is best for consistency.
Anyway I will not insist, so I will change it to your version.

Reply:
If you want to try and build a more general utility function, I agree. But since this regex is being used only for this specific purpose, the simpler the better (without losing robustness ofc)

Reply:
Consistent style was meant for the regex, incl. robustness and the ability to tweak and adapt it for any change without the need to rewrite it from scratch. No utility function seems applicable for that.

In that respect however it would make sense to have versions of `get_element(s)_by...` helpers that would return the matched element rather than its' content.

Reply:
If #2129 is merged, the respective functions can be used to somewhat simplify code such as this:

```patch
diff --git a/yt_dlp/extractor/megatvcom.py b/yt_dlp/extractor/megatvcom.py
index 399e826d6..ced9ba3fb 100644
--- a/yt_dlp/extractor/megatvcom.py
+++ b/yt_dlp/extractor/megatvcom.py
@@ -24,9 +24,7 @@ class MegaTVComBaseIE(InfoExtractor):
     _PLAYER_DIV_ID = 'player_div_id'
 
     def _extract_player_attrs(self, webpage):
-        player_el = self._search_regex(
-            r'(<div[^>]*\sid=(?P<_q>["\'])%s(?P=_q)\s[^>]+>)' %
-            self._PLAYER_DIV_ID, webpage, 'player element')
+        player_el = get_element_html_by_id(self._PLAYER_DIV_ID, webpage)
         return {
             re.sub(r'^data-(?:kwik_)?', '', k): v
             for k, v in extract_attributes(player_el).items()
@@ -68,13 +66,14 @@ class MegaTVComIE(MegaTVComBaseIE):
         _is_article = video_id is None
         webpage = self._download_webpage(url, video_id or display_id)
         if _is_article:
-            video_id = self._search_regex(
-                r'<article[^>]*\sid=["\']Article_(\d+)["\']', webpage, 'article id')
+            article_el = get_element_html_by_attribute('id', r'Article_\d+', webpage, escape_value=False)
+            video_id = extract_attributes(article_el[:article_el.index('>') + 1])['id'].replace('Article_', '')
         player_attrs = self._extract_player_attrs(webpage)
         title = player_attrs.get('label') or self._og_search_title(webpage)
-        description = clean_html(get_element_by_class(
+        description = get_element_by_class(
             'article-wrapper' if _is_article else 'story_content',
-            webpage))
+            webpage)
+        description = clean_html(re.sub(r'<script[^>]*>[^<]+</script>', '', description))
         if not description:
             description = self._og_search_description(webpage)
         thumbnail = player_attrs.get('image') or self._og_search_thumbnail(webpage)
```

Matching article id is admittedly more complicated, since it must be extracted from the attribute value.
The fact that the whole element content is returned (rather than the start tag) may also unexpectedly slightly complicate things, as it is obvious here.

Reply:
I applied the rest but I will not do this part, as it seems more complicated than the original:

```patch
diff --git a/yt_dlp/extractor/megatvcom.py b/yt_dlp/extractor/megatvcom.py
index fa2f06e61..a7dc0fd89 100644
--- a/yt_dlp/extractor/megatvcom.py
+++ b/yt_dlp/extractor/megatvcom.py
@@ -71,8 +71,8 @@ class MegaTVComIE(MegaTVComBaseIE):
         _is_article = video_id is None
         webpage = self._download_webpage(url, video_id or display_id)
         if _is_article:
-            video_id = self._search_regex(
-                r'<article[^>]*\sid=["\']Article_(\d+)["\']', webpage, 'article id')
+            article_el = get_element_html_by_attribute('id', r'Article_\d+', webpage, escape_value=False)
+            video_id = extract_attributes(article_el[:article_el.index('>') + 1])['id'].replace('Article_', '')
         player_attrs = self._extract_player_attrs(webpage)
         title = player_attrs.get('label') or self._og_search_title(webpage)
         description = get_element_by_class(
```